mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,480 @@
+        self.is_binomial = self.num_classes <= 2
+
+        self.metrics["accuracy"] = sk_metrics.accuracy_score(self.y, self.y_pred)
+        self.metrics["example_count"] = len(self.X)
+
+        if predict_proba_fn is not None:
+            self.y_probs = predict_proba_fn(self.X)
+            if self.is_binomial:
+                self.y_prob = self.y_probs[:, 1]
+            else:
+                self.y_prob = None
+        else:
+            self.y_probs = None
+            self.y_prob = None
+
+        if predict_proba_fn is not None:
+            self.metrics['log_loss'] = sk_metrics.log_loss(self.y, self.y_probs)
+
+            if self.is_binomial:
+                self._evaluate_per_class(None, self.y, self.y_pred, self.y_prob)
+                self._log_image_artifact(
+                    lambda: plot_lift_curve(self.y, self.y_probs), "lift_curve",
+                )
+            else:
+                self.metrics['f1_score_micro'] = \
+                    sk_metrics.f1_score(self.y, self.y_pred, average='micro')
+                self.metrics['f1_score_macro'] = \
+                    sk_metrics.f1_score(self.y, self.y_pred, average='macro')
+                for postive_class in range(self.num_classes):
+                    y_per_class = np.where(self.y == postive_class, 1, 0)
+                    y_pred_per_class = np.where(self.y_pred == postive_class, 1, 0)
+                    pos_class_prob = self.y_probs[:, postive_class]
+                    self._evaluate_per_class(
+                        postive_class, y_per_class, y_pred_per_class, pos_class_prob
+                    )
+
+        # TODO: Shall we also log confusion_matrix data as a json artifact ?
+        confusion_matrix = sk_metrics.confusion_matrix(self.y, self.y_pred)
+
+        def plot_confusion_matrix():
+            sk_metrics.ConfusionMatrixDisplay(confusion_matrix=confusion_matrix).plot()
+
+        if hasattr(sk_metrics, 'ConfusionMatrixDisplay'):
+            self._log_image_artifact(
+                plot_confusion_matrix, "confusion_matrix",
+            )
+
+        self._log_metrics()
+        self._log_model_explainability()
+        return EvaluationResult(self.metrics, self.artifacts)
+
+    def _evaluate_regressor(self):
+        self.y_pred = self.model.predict(self.X)
+        self.metrics["example_count"] = len(self.X)
+        self.metrics["mean_absolute_error"] = sk_metrics.mean_absolute_error(self.y, self.y_pred)
+        self.metrics["mean_squared_error"] = sk_metrics.mean_squared_error(self.y, self.y_pred)
+        self.metrics["root_mean_squared_error"] = math.sqrt(self.metrics["mean_squared_error"])
+        self.metrics['sum_on_label'] = sum(self.y)
+        self.metrics['mean_on_label'] = self.metrics['sum_on_label'] / self.metrics["example_count"]
+        self.metrics['r2_score'] = sk_metrics.r2_score(self.y, self.y_pred)
+        self.metrics['max_error'] = sk_metrics.max_error(self.y, self.y_pred)
```

Comment:
we MAY want to guard on this depending on how the tracking server handles floating point errors. max_error can be + / - inf

Reply:
I just tried logging some special value:
```
with mlflow.start_run() as run:
  mlflow.log_metric('inf', np.inf)
  mlflow.log_metric('neg_inf', -np.inf)
  mlflow.log_metric('nan', np.nan)
```
mlflow can display them, shall we do additional handling ?
![image](https://user-images.githubusercontent.com/19235986/146899139-ae9b18c2-b51a-4cc5-bebf-6a743826ef67.png)

Reply:
haha good to know that it can handle those just fine! No concerns from me :)

Reply:
MLflow is built to handle infs and nans. I don't think we need to do anything special here.